pkg/sql/catalog/lease/lease.go

```diff
@@ -1378,7 +1380,9 @@ func (m *Manager) RefreshLeases(ctx context.Context, s *stop.Stopper, db *kv.DB)
 				if evFunc := m.testingKnobs.TestingDescriptorRefreshedEvent; evFunc != nil {
 					evFunc(desc.DescriptorProto())
 				}
-
+			case err := <-rangefeedErrCh:
+				log.Errorf(ctx, "rangefeed on system.descriptors failed: %v; restarting", err)
+				m.watchForUpdates(ctx, descUpdateCh, descDelCh, rangefeedErrCh)
```

Comment:
I think we probably need to do more than just restart the feed here. Namely, it would be nice to refresh our state in case we missed some update that would have resulted in us purging some leases.  I noted the refreshAllLeases channel, but some of the code led me to believe that perhaps that isn't what we want after we've moved to session based leases.  Wondering if someone from foundations might provide some guidance or pick this up. cc @fqazi